desc/objectives/linear_objectives.py

```diff
@@ -403,6 +474,460 @@ def compute(self, params, constants=None):
+        self._args = ["L_lmn"]
+        super().__init__(
+            things=eq, target=target, bounds=bounds, weight=weight, name=name
+        )
+
+    def build(self, eq=None, use_jit=False, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        eq : Equilibrium, optional
+            Equilibrium that will be optimized to satisfy the Objective.
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+        """
+        eq = eq or self.things[0]
+        L_modes = eq.L_basis.modes
+        dim_L = eq.L_basis.num_modes
+
+        if (
+            self.target is None
+        ):  # uses current eq's value of lambda at specified zeta as constraint
+            Lb_lmn, Lb_basis = FourierZernike_to_PoincareZernikePolynomial(
+                eq.L_lmn,
+                eq.L_basis,
+                zeta=self._zeta,
+            )  # Currently this method doesn't support 2 different Poincare BC
+            Lb_modes = Lb_basis.modes
+            self._dim_f = Lb_basis.num_modes
+            self.target = Lb_lmn
+
+        self._A = np.zeros((self._dim_f, dim_L))
+
+        if self._zeta == 0:
+            for i, (l, m, n) in enumerate(Lb_modes):
+                j = np.argwhere(
+                    np.logical_and(
+                        (L_modes[:, :-1] == [l, m]).all(axis=1),
+                        L_modes[:, 2] >= 0,
+                    )
+                )
+                self._A[i, j] = 1
+        elif self._zeta == np.pi:
+            for i, (l, m, n) in enumerate(Lb_modes):
+                j1 = np.argwhere(
+                    np.logical_and(
+                        np.logical_and(
+                            (L_modes[:, :-1] == [l, m]).all(axis=1),
+                            L_modes[:, 2] % 2 == 1,
+                        ),
+                        L_modes[:, 2] >= 0,
+                    )
+                )
+                j2 = np.argwhere(
+                    np.logical_and(
+                        np.logical_and(
+                            (L_modes[:, :-1] == [l, m]).all(axis=1),
+                            L_modes[:, 2] % 2 == 0,
+                        ),
+                        L_modes[:, 2] >= 0,
+                    )
+                )
+                self._A[i, j1] = -1
+                self._A[i, j2] = 1
+        else:
+            raise ValueError(
+                f"Unvalid zeta value! Only 0 or pi is supported! zeta ={self._zeta}"
+                + "is given!"
+            )
+
+        if self.target is not None:
+            self._dim_f = self._A.shape[0]
+
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, params, constants=None):
+        """Compute deviation from desired boundary."""
+        return jnp.dot(self._A, params["L_lmn"])
+
+
+class SecondBoundaryRSelfConsistency(_Objective):
```

Comment:
what are these needed for?

Reply:
I was trying to see if giving 2 Poincare sections (one at zeta=0 and one at zeta=pi) would help or not. Probably these won't be needed in the final PR.